qemu/tests/qmp_command.py

```diff
@@ -19,167 +21,126 @@ def run(test, params, env):
-                if value in exception_list:
-                    continue
-                if value.strip() not in str(qmp_o):
-                    test.fail("QMP command output does not contain "
-                              "expect result. Expect result: '%s'\n"
-                              "Actual result: '%s'" % (value, qmp_o))
-        elif result_check == "not_contain":
-            values = output.split(';')
-            for value in values:
-                if value in exception_list:
-                    continue
-                if value in str(qmp_o):
-                    test.fail("QMP command output contains unexpect"
-                              " result. Unexpect result: '%s'\n"
-                              "Actual result: '%s'" % (value, qmp_o))
-        elif result_check == "m_equal_q":
-            msg = "QMP command ouput is not equal to in human monitor command."
-            msg += "\nQMP command output: '%s'" % qmp_o
-            msg += "\nHuman command output: '%s'" % output
-            res = output.splitlines(True)
-            if type(qmp_o) != type(res):
-                len_o = 1
-            else:
-                len_o = len(qmp_o)
-            if len(res) != len_o:
-                if res[0].startswith(' '):
-                    test.fail("Human command starts with ' ', "
-                              "there is probably some garbage in "
-                              "the output.\n" + msg)
-                res_tmp = []
-                #(qemu)info block in RHEL7 divided into 3 lines
-                for line in res:
-                    if not line.startswith(' '):
-                        res_tmp.append(line)
-                    else:
-                        res_tmp[-1] += line
-                res = res_tmp
-                if len(res) != len_o:
-                    test.fail(msg)
-            re_str = r'([^ \t\n\r\f\v=]*)=([^ \t\n\r\f\v=]*)'
-            for i in range(len(res)):
-                if qmp_cmd == "query-version":
-                    version = qmp_o['qemu']
-                    version = "%s.%s.%s" % (version['major'], version['minor'],
-                                            version['micro'])
-                    package = qmp_o['package']
-                    re_str = r"([0-9]+\.[0-9]+\.[0-9]+)\s*(\(\S*\))?"
-                    hmp_version, hmp_package = re.findall(re_str, res[i])[0]
-                    if not hmp_package:
-                        hmp_package = package
-                    hmp_package = hmp_package.strip()
-                    package = package.strip()
-                    hmp_version = hmp_version.strip()
-                    if version != hmp_version or package != hmp_package:
-                        test.fail(msg)
+            result = True
+            for o in expect_o:
+                if isinstance(o, dict):
+                    for key, val in o.items():
+                        result = result and check_list(qmp_o, key, val)
```

Comment:
if check_list returns false, result becomes false, then we don't need to continue the for loop, right?

Reply:
Would _check_dict() work here instead of check_list in() for loop?

Reply:
We use `check_list` here, because `qmp_o` is a list.   It is not related with `o`,  the type of `o`  is related with the value of `check_item_in_pair` .